pkg/apiserver/registry/stats/nodelatencystats/rest_test.go

```diff
@@ -29,31 +30,59 @@ import (
 )
 
 func TestREST(t *testing.T) {
-	r := NewREST()
+	fakeClock := clocktesting.NewFakeClock(time.Now())
+	r := newRESTWithClock(fakeClock)
 	assert.Equal(t, &statsv1alpha1.NodeLatencyStats{}, r.New())
 	assert.Equal(t, &statsv1alpha1.NodeLatencyStats{}, r.NewList())
 	assert.False(t, r.NamespaceScoped())
 }
 
 func TestRESTCreate(t *testing.T) {
-	summary := &statsv1alpha1.NodeLatencyStats{
-		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
-		PeerNodeLatencyStats: nil,
-	}
-	expectedObj := &statsv1alpha1.NodeLatencyStats{
-		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
-		PeerNodeLatencyStats: nil,
+	fakeClock := clocktesting.NewFakeClock(time.Now())
+	tests := []struct {
+		name        string
+		summary     *statsv1alpha1.NodeLatencyStats
+		expectedObj *statsv1alpha1.NodeLatencyStats
+	}{
+		{
+			name: "create with time added",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: fakeClock.Now()}},
+				PeerNodeLatencyStats: nil,
+			},
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: fakeClock.Now()}},
+				PeerNodeLatencyStats: nil,
+			},
+		},
+		{
+			name: "create with time not added",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: fakeClock.Now().Add(2 * time.Minute)}},
```

Comment:
This is not good because your test case assumes that it will run second and that the clock will have been advanced twice as a result. A key principle of table-driven test cases is that test cases should not depend on each other. They should be runnable individually, independently, in any order, and sometimes even in parallel.

The test should be like this:
```go
func TestRESTCreate(t *testing.T) {
        ctx := context.Background()
        now := time.Now()
        const timeStep = 1*time.Minute
	tests := []struct {
		name        string
		summary     *statsv1alpha1.NodeLatencyStats
		expectedObj *statsv1alpha1.NodeLatencyStats
	}{
		{
			name: "create with existing timestamp",
			summary: &statsv1alpha1.NodeLatencyStats{
				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: now}},
				PeerNodeLatencyStats: nil,
			},
			expectedObj: &statsv1alpha1.NodeLatencyStats{
				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: now}},
				PeerNodeLatencyStats: nil,
			},
		},
		{
			name: "create with no existing timestamp",
			summary: &statsv1alpha1.NodeLatencyStats{
				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
				PeerNodeLatencyStats: nil,
			},
			expectedObj: &statsv1alpha1.NodeLatencyStats{
				// the test case advances the clock by timeStep
				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: now.Add(timeStep)}},
				PeerNodeLatencyStats: nil,
			},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
	                 fakeClock := clocktesting.NewFakeClock(now)
			r := newRESTWithClock(fakeClock)
			fakeClock.SetTime(now.Add(timeStep))
			obj, err := r.Create(ctx, tt.summary, nil, nil)
			require.NoError(t, err)
			assert.Equal(t, tt.expectedObj, obj)
		})
	}
}
```

you can use the code as is. I also updated the test case names for more clarity.